File: Platform/Workspace/Typescript/src/allors/framework/database/data/Result.ts
import { Fetch } from './Fetch';

export class Result {
  public fetchRef: string;

  public fetch: Fetch;

  public name: string;

  public skip: number;

  public take: number;

  constructor(fields?: Partial<Result>) {
    Object.assign(this, fields);
  }

  public toJSON(): any {

    return {
      name: this.name,
      skip: this.skip,
      take: this.take,
      fetchRef: this.fetchRef,
      fetch: this.fetch,
    };

  }
}

File: Platform/Database/Allors.Database.Protocol/Data/Result.cs
// <copyright file="Result.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Data
{
    using System;

    public class Result
    {
        public Guid? FetchRef { get; set; }

        public Fetch Fetch { get; set; }

        public string Name { get; set; }

        public int? Skip { get; set; }

        public int? Take { get; set; }
    }
}
